contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -680,18 +716,53 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
     bytes32[] calldata ss,
     bytes32 rawVs
   ) external override {
-    // TODO: update this function with implementation from the doc
-
     // NOTE: If the arguments to this function are changed, _requireExpectedMsgDataLength and/or
     // TRANSMIT_MSGDATA_CONSTANT_LENGTH_COMPONENT need to be changed accordingly
 
     uint256 initialGas = gasleft(); // This line must come first
+    Report memory report_;
```

Comment:
All execution paths need to decode the report. It might be better to decode at the top, to prevent decoding twice in the case that the secondary feed reports the latest report.

This will change the behaviour of the OCR2Aggregator, since we now decode the report before validating signatures, epoch and round, context... Something worth calling out I think